test/integration/targets/elb_application_lb/tasks/test_modifying_alb_listeners.yml

```
@@ -117,6 +117,81 @@
       that:
         - alb.changed
         - alb.listeners[0].rules|length == 3
+  - name:
+    debug:
+      msg: "{{ alb }}"
+
+  - name: add a rule that uses the host header condition to the listener
+    elb_application_lb:
+      name: "{{ alb_name }}"
+      subnets: "{{ alb_subnets }}"
+      security_groups: "{{ sec_group.group_id }}"
+      state: present
+      purge_rules: no
+      listeners:
+        - Protocol: HTTP
+          Port: 80
+          DefaultActions:
+            - Type: forward
+              TargetGroupName: "{{ tg_name }}"
+          Rules:
+            - Conditions:
+                - Field: host-header
+                  Values:
+                    - 'local.mydomain.com'
+              Priority: '3'
+              Actions:
+                - TargetGroupName: "{{ tg_name }}"
+                  Type: forward
+      <<: *aws_connection_info
+    register: alb
+
+  - name:
+    debug:
+      msg: "{{ alb }}"
+
+  - name:
+    debug:
+      msg: Number of alb rules "{{ alb.listeners[0].rules|length }}"
+
+  - assert:
+      that:
+        - alb.changed
+        - alb.listeners[0].rules|length == 4
```

```suggestion
        - alb.listeners[0].rules|length == 4
        - '{{ alb|json_query("listeners[].rules[].conditions[].host_header_config.values[]")|length  == 1 }}'
```
It would be nice to also assert the expected number of host headers are present.